src/Workspaces/Core/Portable/Workspace/Host/PersistentStorage/IPersistentStorageLocationService.cs

```diff
@@ -58,8 +67,7 @@ protected virtual string GetCacheDirectory()
 
             static string StripInvalidPathChars(string val)
             {
-                var invalidPathChars = Path.GetInvalidPathChars();
-                val = new string(val.Where(c => !invalidPathChars.Contains(c)).ToArray());
+                val = new string(val.Where(c => !s_invalidPathChars.Contains(c)).ToArray());
```

Comment:
It seems you've expanded what this method does. It's now stripping more than what are 'invalid' path chars.
And if this method is special case just for the one purpose, then this seems a really expensive way to go about it. Base64 has only one invalid path char it would ever produce, so couldn't you just say `val.Replace("/","")` which would surely run faster than what you have here?

Reply:
so this runs only once, and i'm ok with the logic as is.  teh general thing here is that we don't know per-se that we're base64 encoding.  that happens to be an impl detail of another api.  so this is basically just saynig: "give us a general hash for yoru name, and we will make it into something that can be used as a path component".  The difference between the old code and the new code was that the old code stripped out things that weren't legal for a path char, while the new code does the same *and* makes sure we're not actually putting in a path separator either.